### ISSUE TITLE
Allure not creating reports 

### ISSUE THREAD

**Issue by RobertoTeresa** [Shelex/cypress-allure-plugin#147]
I am using this plugin with @badeball/cypress-cucumber-preprocessor and it generate the videos and screenshots but not the allure-results

Here is my config.js

```
const { defineConfig } = require("cypress");
const webpack = require("@cypress/webpack-preprocessor");
const preprocessor = require("");
const allureWriter = require('@shelex/cypress-allure-plugin/writer');

async function setupNodeEvents(on, config) {
  await preprocessor.addCucumberPreprocessorPlugin(on, config);
  on(
    "file:preprocessor",
    webpack({
      webpackOptions: {
        resolve: {
          extensions: [".ts", ".js"],
        },
        module: {
          rules: [
            {
              test: /\.feature$/,
              use: [
                {
                  loader: "@badeball/cypress-cucumber-preprocessor/webpack",
                  options: config,
                },
              ],
            },
          ],
        },
      },
    })
  );
  allureWriter(on, config);
  return config;
}

module.exports = defineConfig({
  e2e: {
    specPattern: "**/*.feature",
    chromeWebSecurity: false,
    supportFile: false,
    numTestsKeptInMemory: 3,
    setupNodeEvents,
  },
});


```

And my package
```

{
  "name": "caixa_cypress",
  "version": "1.0.0",
  "description": "Caixa cypress",
  "main": "cypress.config.js",
  "scripts": {
    "test": "cypress open"
  },
  "author": "Roberto Teresa",
  "license": "ISC",
  "devDependencies": {
    "@badeball/cypress-cucumber-preprocessor": "^11.4.0",
    "@bahmutov/cypress-esbuild-preprocessor": "^2.1.3",
    "@cypress/browserify-preprocessor": "^3.0.2",
    "@cypress/webpack-preprocessor": "^5.12.0",
    "@shelex/cypress-allure-plugin": "^2.28.0",
    "allure-commandline": "^2.18.1",
    "cypress": "^10.3.0",
    "cypress-iframe": "^1.0.1",
    "cypress-xpath": "^2.0.0",
    "esbuild": "^0.14.48",
    "mocha-allure-reporter": "^1.4.0"
  },
  "dependencies": {
    "cypress-real-events": "^1.7.1"
  }
}
```

**Comment by Shelex** [comment]
Hi @RobertoTeresa 
Thank you for opening an issue.

I would like to clarify:
- have you imported `cypress-allure-plugin` commands in `support/index` file?
- what is the full command you use for running tests with report generation? as in your `package.json` I see just `cypress open`.

Thanks!

**Comment by GRmeB** [comment]
I have the same issue. The allure JSONs are not generated anymore for me since the new version 2.28.1.

I'm calling the tests as the same as before:

`./node_modules/.bin/cypress run --env allure=true --browser chrome --config-file config/cypress_stage.ts --spec cypress/e2e/mocha/functional/mytest.spec.js`

And I've already imported the plugin in `support/index.ts`.

My config looks like this:

```
export async function setupNodeEvents(
    on: Cypress.PluginEvents,
    config: Cypress.PluginConfigOptions
): Promise<Cypress.PluginConfigOptions> {
    //await addCucumberPreprocessorPlugin(on, config);

    //on(
    //    'file:preprocessor',
    //    browserify(config, {
    //        typescript: require.resolve('typescript'),
    //    })
    //);

    const options = {
        ...browserify.defaultOptions,
        typescript: path.join(path.resolve('..'), 'node_modules/typescript'),
    };

    on('file:preprocessor', await preprocessor(config, options));

    await addCucumberPreprocessorPlugin(on, config);

    require('cypress-grep/src/plugin')(config);
    require('cypress-log-to-output').install(on);
    require('@cypress/code-coverage/task')(on, config);
    config = dotenvPlugin(config);

    if (config.env.USE_MOCHA === true) {
        config.specPattern = 'cypress/e2e/mocha/**/*.js';
    }
    allureWriter(on, config);

    return config;
}

module.exports = defineConfig({
// some more code  
    e2e: {
        specPattern: 'cypress/e2e/**/*.feature',
        experimentalSessionAndOrigin: false,
        setupNodeEvents,
    },
});
```

**Comment by javierfuentesm** [comment]
I have the same issue when i do this : 

`  on(
    "file:preprocessor",
    webpack({
      webpackOptions: {
        resolve: {
          extensions: [".ts", ".js"],
        },
        module: {
          rules: [
            {
              test: /\.feature$/,
              use: [
                {
                  loader: "@badeball/cypress-cucumber-preprocessor/webpack",
                  options: config,
                },
              ],
            },
          ],
        },
      },
    })
  );
  AllureWriter(on, config);`

it fails but when i do this :  
` on("file:preprocessor", webpack);
  AllureWriter(on, config);
`

it generates the results but the tests are not being parsed correctly 
<img width="758" alt="image" src="https://user-images.githubusercontent.com/19244527/182769737-f998ee9b-6a16-43a0-a22b-633295f2844c.png">


Any ideas??

**Comment by javierfuentesm** [comment]
@RobertoTeresa  did you were able to solve it ??

**Comment by Shelex** [comment]
@javierfuentesm 
do you return `config` after mutating it?
```js
on("file:preprocessor", webpack); 
AllureWriter(on, config);
return config;
```

**Comment by javierfuentesm** [comment]
@Shelex yes i am

**Comment by Shelex** [comment]
So, actually I have no idea what may be the issue as I cannot reproduce it with exact outcomes mentioned here.
I am playing with [test repo](https://github.com/Shelex/cypress-cucumber-allure) where I have it working with cucumber preprocessor.

I tried to replicate @GRmeB setup - [plugins-file](https://github.com/Shelex/cypress-cucumber-allure/blob/d0a1fd24dc16fa8d421537f69940a4202b9e859f/cypress/plugins/index.ts), and I have no issue with reporting or cypress, but with webpack compilation - [link to branch](https://github.com/Shelex/cypress-cucumber-allure/tree/issue/GRmeB#issue).

As for @javierfuentesm case - [plugins-file](https://github.com/Shelex/cypress-cucumber-allure/blob/0b8701103c1f9858410a973d18cdbd3e4c08593d/cypress/plugins/index.ts) I have it working ¯\_(ツ)_/¯ - [link to branch](https://github.com/Shelex/cypress-cucumber-allure/tree/issue/webpack-actually-it-works#outcome).

I see that all these cases have in common the cucumber-preprocessor, so I would strongly recommend to check their setup docs with bundler of your choice (webpack, browserify, whatever), validate it is working. Also you can upgrade versions - maybe there were some issues fixed, who knows.

In case it reproduces on your side - please prepare an example repository, that will help to understand your specific case.

**Comment by javierfuentesm** [comment]
@Shelex  thanks for all your help , i checked and the config is the same :( https://github.com/javierfuentesm/CypressAvanzadoPlatzi/tree/21/reportes
Here my branch i dont see any difference with what you have , do you any more ideas?

**Comment by Shelex** [comment]
> @Shelex thanks for all your help , i checked and the config is the same :( https://github.com/javierfuentesm/CypressAvanzadoPlatzi/tree/21/reportes Here my branch i dont see any difference with what you have , do you any more ideas? thank you again

Thanks for example - can confirm it reproduces. However, I can see that reporter is not even initialized, as `Cypress.Allure` is `undefined`, but according to logs `writer` part is fine, it just do not receive any information from browser:
```
allure-plugin:writer inside "after:spec" event +8s
allure-plugin:writer processing "after:spec" results +0ms
allure-plugin:writer not found mapping of mocha test id to allure test ids +0ms
```

All I can add is that it needs more investigation.

**Comment by javierfuentesm** [comment]
@Shelex  thanks for all the help

**Comment by javierfuentesm** [comment]
@Shelex i was doing some debugging and i removed the webpack processor and left  only the allure report and tried to run a normal test, and it didnt work :/ it is no weird im wondering if the reason is that you use typescript and I use js

**Comment by javierfuentesm** [comment]
@Shelex  ok , I realized that when i comment this line 
<img width="326" alt="image" src="https://user-images.githubusercontent.com/19244527/182991382-ff34fbe5-c24a-43c9-b2c6-d641c96a4e81.png">
it works without cucumber  
but when i leave it  now i have this error 
<img width="1390" alt="image" src="https://user-images.githubusercontent.com/19244527/182991452-f0f11d20-8f4a-4578-a25e-bc0bfe019c88.png">

and this is my file 
<img width="616" alt="image" src="https://user-images.githubusercontent.com/19244527/182991486-217bc189-3d6f-4497-86f6-da7d53c102bf.png">

Any ideas? i dont know why in your project works fine , i even commented all the other lines and it fails 
<img width="500" alt="image" src="https://user-images.githubusercontent.com/19244527/182991766-058fcce3-6f7e-4dc1-9c9d-aee1a9f0b9d2.png">
<img width="1220" alt="image" src="https://user-images.githubusercontent.com/19244527/182991846-c707dbce-4064-4614-84e9-4d1659870dd8.png">

**Comment by javierfuentesm** [comment]
@Shelex ok so I solved it by doing this:
adding the fallback 
<img width="619" alt="image" src="https://user-images.githubusercontent.com/19244527/182997141-88bc5fe6-0396-4cd2-8a3a-92803c6fe8ea.png">

and it works !!
i think the confusion comes , when you follow the example of this repo https://github.com/badeball/cypress-cucumber-preprocessor/blob/9dc9938c7d0d615b4110459af1710ed63dfe6f44/examples/webpack-cjs/cypress.config.js 
and it has the supportFile: false property 
Thanks for the patience

**Comment by Shelex** [comment]
@javierfuentesm Great catch!
Indeed, `"supportFile": false` was an elephant in a room, which actually blocks anything that requires imports in `support` file to work (as support file is not even bundled).

Regarding `util` package issue - webpack v5+ do not provide any polyfils for packages that are from node library. As I use `util.inspect` just for pretty-printing objects, I have switched to `object-inspect` library and published `v2.28.4`.
I have also checked your branch locally with latest plugin version - seems to be working, but I would be glad if you check that also.
